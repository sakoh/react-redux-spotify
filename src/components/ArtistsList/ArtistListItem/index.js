import React from 'react'
import { Avatar } from '@material-ui/core/Avatar'
import { ListItem } from '@material-ui/core/ListItem'
import { ListItemText } from '@material-ui/core/ListItemText'
import PropTypes from 'prop-types'
import { artistPropType } from '../types'

const ArtistListItem = ({
  name,
  image,
  popularity
}) => (
  <ListItem>
    <Avatar src={image} />
    <ListItemText primary={name} secondary={popularity} />
  </ListItem>  
)

ArtistListItem.propTypes = artistPropType

export default ArtistListItem
