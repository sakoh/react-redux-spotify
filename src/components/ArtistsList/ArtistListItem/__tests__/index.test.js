import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import PropTypes from 'prop-types'
import ArtistListItem from '..'

configure({ adapter: new Adapter() })

describe('ArtistsListItem', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <ArtistListItem
        name='Iron Maiden'
        image='http://via.placeholder.com/350x150'
        popularity={10} 
      />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
