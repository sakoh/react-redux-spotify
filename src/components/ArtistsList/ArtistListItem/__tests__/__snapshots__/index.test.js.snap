// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistsListItem should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArtistListItem
    image="http://via.placeholder.com/350x150"
    name="Iron Maiden"
    popularity={10}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Unknown
          src="http://via.placeholder.com/350x150"
/>,
        <Unknown
          primary="Iron Maiden"
          secondary={10}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "src": "http://via.placeholder.com/350x150",
        },
        "ref": null,
        "rendered": null,
        "type": undefined,
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "primary": "Iron Maiden",
          "secondary": 10,
        },
        "ref": null,
        "rendered": null,
        "type": undefined,
      },
    ],
    "type": undefined,
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Unknown
            src="http://via.placeholder.com/350x150"
/>,
          <Unknown
            primary="Iron Maiden"
            secondary={10}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "src": "http://via.placeholder.com/350x150",
          },
          "ref": null,
          "rendered": null,
          "type": undefined,
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "primary": "Iron Maiden",
            "secondary": 10,
          },
          "ref": null,
          "rendered": null,
          "type": undefined,
        },
      ],
      "type": undefined,
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
